guppylang/compiler/core.py

```diff
@@ -48,6 +54,16 @@ def _compile(self, defn: CheckedDef) -> CompiledDef:
             return defn.compile_outer(self.module)
         return defn
 
+    def get_instance_func(
+        self, ty: Type | TypeDef, name: str
+    ) -> CompiledCallableDef | None:
+        checked_func = self.checked_globals.get_instance_func(ty, name)
```

Comment:
This does feel like some refactoring could help, perhaps after the array stuff is all merged in; all we actually want from the checked-globals is the mapping from name to id...

Reply:
We also need the mapping from class definitions to their methods